### ISSUE TITLE
SDWebImage (5.3.1), iOS 12.0.1 version, error loading animation image

### ISSUE THREAD

**Issue by SarielTang** [SDWebImage/SDWebImage#2991]
Found on an iPhone7 iOS12.0.1 mobile phone, there was a problem loading the picture below.
![1](https://user-images.githubusercontent.com/12050371/80199875-62bab700-8654-11ea-930f-d4a8a5d91a47.png)
The scenario is as follows
![111](https://user-images.githubusercontent.com/12050371/80200677-91855d00-8655-11ea-974d-b2ec6c42d8eb.gif)

**Comment by dreampiggy** [comment]
APNG ?

Using which View, UIImageView or SDAnimatedImageView ?

**Comment by SarielTang** [comment]
> @SarielTang APNG ?
> 
> Using which View, UIImageView or SDAnimatedImageView ?

SDAnimatedImageView

**Comment by dreampiggy** [comment]
The latest master branch and v5.7.3 version works without any rendering issue.
[master.mov.zip](https://github.com/SDWebImage/SDWebImage/files/4528251/master.mov.zip)

**Comment by dreampiggy** [comment]
Are you have some extra CALayer construction ? I guess that blue strange thing is from layer composition.

**Comment by dreampiggy** [comment]
And , you can try to directlly upgrade to latest version to have a check again. SDWebImage follow semantic version, there are no API break in 5.x.x any version.

**Comment by SarielTang** [comment]
> And , you can try to directlly upgrade to latest version to have a check again. Even it's a bug, we still fix it in the master branch and release new 5.7.4, not a 5.3.2.
> 
> SDWebImage follow semantic version, there are no API break in 5.x.x any version. it's always safe to upgrade.

Okay, thank you, I will try to upgrade later. This problem is not necessarily reproducible, but there is a problem on the iPhone7 iOS12.0.1 model.

**Comment by dreampiggy** [comment]
Can Xcode's simulator reproduce this ?

I guess this is a bug from Apple's Image/IO framework itself. The CGImage which may contains wrong background. No extra workaround, until we re-implements the APNG decoding itself.

Or, can you hava a try by using the [SDWebImageYYPlugin](https://github.com/SDWebImage/SDWebImageYYPlugin). Which have a bridge to `YYImageDecoder`, it contains a APNG decoder written by libpng, not Apple's ImageIO.

For example, use the code:

```objective-c
  [SDImageCodersManager removeCoder:SDImageAPNGCoder.sharedCoder];
// Check APNG bug
if (@available(iOS, 13.0, *)) {
  [SDImageCodersManager addCoder:SDImageAPNGCoder.sharedCoder];
} else if (@available(iOS, 12.0, *)) {
    // Image/IO Bug !!!, Use libpng for APNG decoding
    [SDImageCodersManager addCoder:YYImageDecoder.sharedCoder];
} else {
  [SDImageCodersManager addCoder:SDImageAPNGCoder.sharedCoder];
}
```

**Comment by SarielTang** [comment]
After upgrading the version, the problem is solved, thank you very much
